datafusion/core/src/physical_optimizer/sort_pushdown.rs

```diff
@@ -0,0 +1,394 @@
+// Licensed to the Apache Software Foundation (ASF) under one
```

Comment:
This file is pruned version of top-down rule by @mingmwang at #5290. Its scope is converted from top-down sort enforcement to sort pushdown.